clients/rust/tests/harvest_holder_rewards.rs

```diff
@@ -0,0 +1,783 @@
+    holder_rewards_account.last_accumulated_rewards_per_token = 40_000_000 * 1_000_000_000;
+
+    account.data = holder_rewards_account.try_to_vec().unwrap();
+    context.set_account(&holder_rewards, &account.into());
+
+    // And we harvest the holder rewards.
+
+    let destination = Pubkey::new_unique();
+
+    let harvest_holder_rewards_ix = HarvestHolderRewardsBuilder::new()
+        .config(config_manager.config)
+        .stake(stake_manager.stake)
+        .vault(config_manager.vault)
+        .holder_rewards(holder_rewards)
+        .vault_authority(find_vault_pda(&config_manager.config).0)
+        .stake_authority(stake_manager.authority.pubkey())
+        .destination(destination)
+        .mint(config_manager.mint)
+        .token_program(spl_token_2022::ID)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[harvest_holder_rewards_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &stake_manager.authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    let account = get_account!(context, destination);
+    assert_eq!(account.lamports, 2_000_000_000);
+
+    // When we harvest the holder rewards again.
+
+    let second_destination = Pubkey::new_unique();
+
+    let harvest_holder_rewards_ix = HarvestHolderRewardsBuilder::new()
+        .config(config_manager.config)
+        .stake(stake_manager.stake)
+        .vault(config_manager.vault)
+        .holder_rewards(holder_rewards)
+        .vault_authority(find_vault_pda(&config_manager.config).0)
+        .stake_authority(stake_manager.authority.pubkey())
+        .destination(second_destination)
+        .mint(config_manager.mint)
+        .token_program(spl_token_2022::ID)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[harvest_holder_rewards_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &stake_manager.authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the second destination account shoud have no rewards.
+
+    let account = context
+        .banks_client
+        .get_account(second_destination)
+        .await
+        .unwrap();
+    assert!(account.is_none());
+
+    // And there should be rewards left on the vault account.
+
+    let account = get_account!(context, config_manager.vault);
+    assert_eq!(
+        account.lamports,
+        rewards_lamports.saturating_sub(2_000_000_000)
+    );
+
+    // And the stake account last seen holder rewards per token is update.
+
+    let account = get_account!(context, stake_manager.stake);
+    let stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(
+        stake_account.last_seen_holder_rewards_per_token,
+        40_000_000 * 1_000_000_000
+    );
```

Comment:
This gets set from the first harvest, though, right? The second one should see zero rewards eligible and terminate early?

Reply:
Yes, the test is asserting that the value of `last_seen_holder_rewards_per_token` did not change from the first harvest.